### PR TITLE
AZ-62: Add synchronization workflow with foundation repository

### PR DESCRIPTION
**Pull request by maciejnems**
I tested it here:
* workflow - https://github.com/Cardinal-Cryptography/aleph-node/runs/4187960088?check_suite_focus=true
* commit - https://github.com/Cardinal-Cryptography/aleph-node/commit/bba0b589d47c2786167ab00366065832accd1bd4
* foundation branch - https://github.com/aleph-zero-foundation/aleph-node/tree/AZ-62-add-sync-with-foundation-repo

### CODE REVIEW

File: .github/workflows/push-foundation-repo.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    branches: [main]
+
+jobs:
+  sync:
+    runs-on: ubuntu-latest
+    if: ${{ github.repository == 'Cardinal-Cryptography/aleph-node'}}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```
Comment (kostekIV) [inline .github/workflows/push-foundation-repo.yml:14]: What is the reason for `fetch-depth` set to 0?

Comment (maciejnems) [reply .github/workflows/push-foundation-repo.yml:14]: By default checkout@v2 fetches only one commit, so the push is not possible. We need to fetch the full history

**Review by kostekIV — APPROVED** [review]

File: .github/workflows/push-foundation-repo.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+          token: ${{ secrets.SYNCAZF }}
```
Comment (Yithis) [inline .github/workflows/push-foundation-repo.yml:15]: There is the same token for both AlephBFT and aleph-node?

Comment (maciejnems) [reply .github/workflows/push-foundation-repo.yml:15]: Yes. More specifically this token has write rights to aleph-foundation organization repos.

**Review by Yithis — APPROVED** [review]

File: .github/workflows/push-foundation-repo.yml (hunk 2 above)
Comment (Yithis) [reply .github/workflows/push-foundation-repo.yml:15]: I'm not sure if it is safe to have one token for all organization repos, but won't also fight for changing it

Comment (maciejnems) [reply .github/workflows/push-foundation-repo.yml:15]: I don't think there is a problem here. We keep all our work in Cardinal Cryptography organization, so having write token for repos in Aleph Foundation organization won't mess with our work in any way

**Pull request merged**